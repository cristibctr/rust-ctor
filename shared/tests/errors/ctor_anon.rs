
shared::ctor_parse!(
    #[ctor(anonymous)]
    unsafe fn foo() {
        println!("Hello, world!");
    }
);

shared::ctor_parse!(
    #[ctor]
    unsafe fn bar() {
        println!("Hello, world!");
    }
);

fn main() {
    // Disallowed
    foo();
    // Allowed
    bar();
}
