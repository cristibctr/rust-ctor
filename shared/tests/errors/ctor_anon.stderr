error[E0425]: cannot find function `foo` in this scope
  --> tests/errors/ctor_anon.rs:18:5
   |
18 |     foo();
   |     ^^^ not found in this scope
